pkg/config/mesh/kubemesh/watcher.go

```diff
@@ -44,6 +44,8 @@ func NewConfigMapWatcher(client kube.Client, namespace, name, key, extraConfig s
 			log.Warnf("failed to read mesh config from ConfigMap: %v", err)
 			return
 		}
+		w.Mutex.Lock()
```

Comment:
Is the lock here needed? `HandleMeshConfig` is already locking ... or am I missing something?

Reply:
Only if we make two variants of HandleMeshConfig. It would work here, but in HandleUserMeshConfig we need to do some work and then call HandleMeshConfig with the lock held the entire time

Reply:
I agree with @nmittler, using a lock to protect a calling method is hard to develop in future.

Reply:
I have addressed the concern, locking is now handled internally